### PR TITLE
JS: make the JS autobuilder consistent with Ruby when no JS code was detected

### PR DESCRIPTION
**Pull request by erik-krogh**
See the backref for some context.

### CODE REVIEW

**Comment by calumgrant** [comment]
The thing I'm a bit concerned about here is if deleting the source directory could have other consequences in non-standard setups, where some other extraction step or invocation populated this directory.

Is there really a need to recursively delete it if we skip `extractExterns()`?

**Comment by erik-krogh** [comment]
> Is there really a need to recursively delete it if we skip extractExterns()? 

Yes, because the folder was created (but empty).  
I've changed it such that the folder is only deleted if it is empty.

File: javascript/extractor/src/com/semmle/js/extractor/AutoBuild.java [hunk 1]
```diff
@@ -433,23 +434,40 @@ private boolean addPathPattern(Set<Path> patterns, Path base, String pattern) {
...
-    if (!seenCode) {
+    if (!hasSeenCode()) {
       if (seenFiles) {
         warn("Only found JavaScript or TypeScript files that were empty or contained syntax errors.");
       } else {
         warn("No JavaScript or TypeScript code found.");
       }
-      return -1;
+      // ensuring that the finalize steps detects that no code was seen. 
+      Path srcFolder = Paths.get(EnvironmentVariables.getWipDatabase() + "/src");
```
Comment (asgerf) [inline javascript/extractor/src/com/semmle/js/extractor/AutoBuild.java:464]:
```suggestion
      Path srcFolder = Paths.get(EnvironmentVariables.getWipDatabase(), "src");
```

**Review by asgerf — COMMENTED** [review]
Let's also update extractor version string, and run DCA.

**Comment by erik-krogh** [comment]
The evaluations are weird, for some reason the externs aren't extracted in some of the projects, even though there are files present, and `js/syntax-error` is empty.  

The generated databases are also of very different sizes.  
Example: https://github.com/dsp-testing/codeql-dca-worker/actions/runs/3586515053 

I'm investigating some more.

**Comment by erik-krogh** [comment]
And I finally got [the latest evaluation](https://github.com/github/codeql-dca-main/tree/data/erik-krogh/pr-11472-6289ae__nightly-old__code-scanning/reports) to look good.   
There is a bunch of noise I can't explain.  
But there is no alert difference, and no performance regression.

**Comment by asgerf** [comment]
The changes in database sizes look a bit suspicious. Could you double-check one of those to see what's going on?

**Comment by erik-krogh** [comment]
> The changes in database sizes look a bit suspicious. Could you double-check one of those to see what's going on?

It's only the "including cache" summary that goes wrong.  
The `## Database size after brutal cleanup, per source` summary is clean, there is a difference of exactly 0 in all databases.

**Comment by erik-krogh** [comment]
[Another evaluation on default.yml](https://github.com/github/codeql-dca-main/tree/data/erik-krogh/pr-11472-6289ae__nightly-old__code-scanning__1/reports) shows the same benign pattern.

**Review by asgerf — APPROVED** [review]
👍

**Pull request merged**